pkg/trait/container.go

```diff
@@ -297,17 +298,19 @@ func (t *containerTrait) configureContainer(e *Environment) error {
 		return err
 	}
 
-	// CronJob
-	if err := e.Resources.VisitCronJobE(func(cron *v1beta1.CronJob) error {
-		for _, envVar := range e.EnvVars {
-			envvar.SetVar(&container.Env, envVar)
-		}
+	if ok, err := kubernetes.IsAPIResourceInstalled(e.Client, batchv1.SchemeGroupVersion.String(), reflect.TypeOf(batchv1.CronJob{}).Name()); ok && err == nil {
```

Comment:
Is this really necessary? My understanding is that there is no server API requests made while visiting the resources?

Reply:
yes let's try without. makes life in tests much easier